### DISCUSSION TITLE
Changing the endianness of the Modbus client?

### DISCUSSION THREAD

**Discussion by nikstp** [pymodbus-dev/pymodbus#1248] [Help] [answered]
Hi, 

I've been having some issues with converting little endian to big endian floats I get from the Modbus client. I know there are programmatic workarounds, but looking through the documentation, it seems there are functions or ways to set the client or do that "natively", that is using Pymodbus's internal functions or objects. Am I correct and if I am, could you point me to an example which does that or where should I look? 

Thank you!

**Comment by janiversen** [comment, marked as answer]
There are no way to tell the client to deliver in little endian for floats, since floats are minimum 2 registers. 

When you read a register with pymodbus the 2 bytes are automatically set to the endianness of the CPU, what you need to do is to combine reg[0] + reg[1] or reg[1] + reg[0] depending on your CPU.

in https://github.com/riptideio/pymodbus/blob/dev/pymodbus/datastore/simulator.py we have internal function to do the changes:
```
@classmethod
    def build_value_from_registers(cls, registers, is_int):
        """Build registers from int32 or float32"""
        value_bytes = int.to_bytes(registers[0], 2, "big") + int.to_bytes(
            registers[1], 2, "big"
        )
        if is_int:
            value = int.from_bytes(value_bytes, "big")
        else:
            value = struct.unpack("f", value_bytes)[0]
        return value
```

**Reply by nikstp** [reply]
Okay, thank you, I'll check that example out!
